lib/common/xml.c

```diff
@@ -1848,47 +1848,63 @@ pcmk__xml2text(const xmlNode *data, uint32_t options, GString *buffer,
     }
 }
 
-char *
-dump_xml_formatted_with_text(const xmlNode *xml)
+/*!
+ * \internal
+ * \brief Dump an XML tree to a string
+ *
+ * \param[in] xml    XML tree to dump
+ * \param[in] flags  Group of <tt>enum pcmk__xml_fmt_options</tt> flags
+ *
+ * \return Newly allocated string representation of \p xml
+ *
+ * \note The caller is responsible for freeing the return value using
+ *       \c g_free().
+ */
+gchar *
+pcmk__xml_dump(const xmlNode *xml, uint32_t flags)
 {
-    /* libxml's xmlNodeDumpOutput() would work here since we're not specifically
-     * filtering out any nodes. However, use pcmk__xml2text() for consistency,
-     * to escape attribute values, and to allow a const argument.
+    /* libxml2's xmlNodeDumpOutput() doesn't allow filtering, doesn't escape
+     * special characters thoroughly, and doesn't allow a const argument.
+     *
+     * @COMPAT Can we start including text nodes unconditionally?
```

Comment:
You can move the compat comment to the `pcmk__xml_fmt_text` definition. The rest applies equally to `pcmk__xml2text()`